spec.md

```diff
@@ -2948,38 +2941,52 @@ The interpret as SInt operation reinterprets e's bits as a signed integer accord
+| -------- | ---------- | ----------- | ------------- | ------------ | -------------|
+| asReset  | \(e\)      | \(t\)       | (Reset)       | t            | n/a          |
+|          |            |             | (UInt<1>)     | t            | n/a          |
+|          |            |             | (SInt<1>)     | t            | n/a          |
+|          |            |             | (Clock)       | t            | n/a          |
+
+The result of the interpret as reset operation is an reset typed signal.  This operation doesn't interpret the data based on type, it is a simple cast.  To write general reset operations, use the `setReset`{.firrtl} operation.
 
-  -----------------------------------------------------------------------------------
-  Name           Arguments   Parameters   Arg Types      Result Type   Result Width
-  -------------- ----------- ------------ -------------- ------------- --------------
-  asAsyncReset   \(e\)       ()           (AsyncReset)   AsyncReset    n/a
 
-                                          (UInt)         AsyncReset    n/a
+## Activate or Release a Reset
 
-                                          (SInt)         AsyncReset    n/a
+|  Name    | Arguments  | Parameters  | Arg Types  | Result Type  | Result Width |
+| -------- | ---------- | ----------- | ---------- | ------------ | -------------|
+| setReset | \(e\)      | \(t\)       | (UInt<1>)  | t            | n/a          |
 
-                                          (Interval)     AsyncReset    n/a
+This operation takes a single bit value and interprets a 1 as an active reset and interprets a 0 as an inactive reset.  This interpretation of the argument is mapped to the appropriate values for the result type.
 
-                                          (Clock)        AsyncReset    n/a
+This allows writing generic reset handling code by decoupling the reset implementation from activation logic.
 
-                                          (Reset)        AsyncReset    n/a
-  -----------------------------------------------------------------------------------
+## Test a Reset
 
-The result of the interpret as asynchronous reset operation is an AsyncReset typed signal.
+|  Name     | Arguments  | Arg Types  | Result Type  | Result Width |
+| --------- | ---------- | ---------- | ------------ | -------------|
+| testReset | \(e\)      | (Reset)    | (UInt<1>)    | 1            |
+
+This operation returns a 1 if a reset is active and a 0 if a reset is inactive regardless of the type of the reset signal.
+
+This allows writing generic reset handling code by decoupling the reset implementation from activation logic.  This operation is different from `asUInt`{.firrtl} in that it always returns `1`{.firrtl} for active resets regardless of the underlying implmentation values.
+
+The behavior of a circuit testing an asynchronous reset are unspecified due to the lack of
+an execution model for FIRRTL in the presence of signal transitions between clock
```

Comment:
I fully agree with this. But it's also kind of telling on ourselves :)

I think it's sufficient for us to just say, "the behavior is unspecified". We can then leave the execution model as an aspiration.